From f9dee3ae0d9bb87c475f144c115f9f780e43b926 Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Mon, 4 Mar 2019 04:14:25 -0800
Subject: [PATCH 0535/1851] drm: xlnx: sdi: Correcting drm property creation
 logic

commit 58b43164ae0ad039bd9f55e08725328a7bf5b308 from
https://github.com/Xilinx/linux-xlnx.git

4.19 kernel adds extra check for flag validity in drm property creation.
Modified the arguments to create drm properties without fail.

Signed-off-by: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_sdi.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_sdi.c b/drivers/gpu/drm/xlnx/xlnx_sdi.c
index 3b26e9287d2b..e97cb43abe44 100644
--- a/drivers/gpu/drm/xlnx/xlnx_sdi.c
+++ b/drivers/gpu/drm/xlnx/xlnx_sdi.c
@@ -646,13 +646,13 @@ xlnx_sdi_drm_connector_create_property(struct drm_connector *base_connector)
 	struct drm_device *dev = base_connector->dev;
 	struct xlnx_sdi *sdi  = connector_to_sdi(base_connector);
 
-	sdi->is_frac_prop = drm_property_create_bool(dev, 1, "is_frac");
+	sdi->is_frac_prop = drm_property_create_bool(dev, 0, "is_frac");
 	sdi->sdi_mode = drm_property_create_range(dev, 0,
 						  "sdi_mode", 0, 5);
 	sdi->sdi_data_strm = drm_property_create_range(dev, 0,
 						       "sdi_data_stream", 2, 8);
-	sdi->sdi_420_in = drm_property_create_bool(dev, 1, "sdi_420_in");
-	sdi->sdi_420_out = drm_property_create_bool(dev, 1, "sdi_420_out");
+	sdi->sdi_420_in = drm_property_create_bool(dev, 0, "sdi_420_in");
+	sdi->sdi_420_out = drm_property_create_bool(dev, 0, "sdi_420_out");
 	sdi->height_out = drm_property_create_range(dev, 0,
 						    "height_out", 2, 4096);
 	sdi->width_out = drm_property_create_range(dev, 0,
@@ -662,9 +662,9 @@ xlnx_sdi_drm_connector_create_property(struct drm_connector *base_connector)
 	sdi->out_fmt = drm_property_create_range(dev, 0,
 						 "out_fmt", 0, 16384);
 	if (sdi->enable_st352_chroma) {
-		sdi->en_st352_c_prop = drm_property_create_bool(dev, 1,
+		sdi->en_st352_c_prop = drm_property_create_bool(dev, 0,
 								"en_st352_c");
-		sdi->use_ds2_3ga_prop = drm_property_create_bool(dev, 1,
+		sdi->use_ds2_3ga_prop = drm_property_create_bool(dev, 0,
 								 "use_ds2_3ga");
 	}
 }
-- 
2.31.1

